xlators/features/gfid-access/src/gfid-access.c

```diff
@@ -543,8 +544,10 @@ ga_heal_entry(call_frame_t *frame, xlator_t *this, loc_t *loc, data_t *data,
 
     if (!xdata)
         xdata = dict_new();
-    else
+    else {
         xdata = dict_ref(xdata);
+        xdata_ref = _gf_true;
```

Comment:
You should update xdata_ref = _gf_true after line 556

Reply:
done